gr-zeromq/lib/base_impl.cc

```diff
@@ -54,22 +54,52 @@ namespace gr {
         delete d_context;
     }
 
-
-    base_sink_impl::base_sink_impl(int type, size_t itemsize, size_t vlen, char *address, int timeout, bool pass_tags, int hwm)
-        : base_impl(type, itemsize, vlen, timeout, pass_tags)
+    void
+    base_impl::set_hwm()
     {
       /* Set high watermark */
-      if (hwm >= 0) {
-#ifdef ZMQ_SNDHWM
-        d_socket->setsockopt(ZMQ_SNDHWM, &hwm, sizeof(hwm));
+      if (d_hwm >= 0) {
+#ifdef ZMQ_RCVHWM
+        d_socket->setsockopt(ZMQ_RCVHWM, &d_hwm, sizeof(d_hwm));
 #else // major < 3
-        uint64_t tmp = hwm;
+        uint64_t tmp = d_hwm;
         d_socket->setsockopt(ZMQ_HWM, &tmp, sizeof(tmp));
 #endif
       }
+    }
+
+    std::string
+    base_impl::endpoint()
+    {
+      size_t len;
+      char endpoint[1024];
+      d_socket->getsockopt(ZMQ_LAST_ENDPOINT, endpoint, &len);
```

Comment:
Going into getsockopt() you have to have len set to the size of the endpoint buffer, then it modifies len on the wait out with the actual size of the data it fills it with.  So you need to have len = 1024 and define endpoint with the variable.  And I think this is done differently in gcc vs MSVC.